espnet2/asr/frontend/adapter_utils/add_adapters.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+from espnet2.asr.frontend.adapter_utils.adapters.adapter_transformer import (
+    AdapterTransformerSentenceEncoderLayer,
+)
+
+
+def add_adapters(s3prl_upstream_model, adapter_down_dim, adapt_layers=None):
+    if adapt_layers == []:
+        print(">> adapt_layers is an empy list. No adapters will be inserted.")
+        return s3prl_upstream_model
+    orig_param_num = count_params(s3prl_upstream_model)
+
+    # freeze all layers
+    for param in s3prl_upstream_model.parameters():
+        param.requires_grad = False
+
+    for layer_idx, layer in enumerate(s3prl_upstream_model.model.encoder.layers):
```

Comment:
`model.encoder.layers` is very specific. What could happen to an upstream with a different model level?